### ISSUE TITLE
Tagger.tag segfaults if Tagger.open is not called

### ISSUE THREAD

**Issue by kmike** [tpeng/python-crfsuite#28]

**Comment by Ridinstar** [comment]
I also had similar issue, I had segfault when I use tagger.tag(), I did call tagger.open(modelfile)

**Comment by kmike** [comment]
@Ridinstar if you did call tagger.open it is likely a different issue. Do you have an example, how to reproduce it?

**Comment by Ridinstar** [comment]
@kmike I was working on ACE2005 data. I was following this tutorial http://nbviewer.jupyter.org/github/tpeng/python-crfsuite/blob/master/examples/CoNLL%202002.ipynb
So I made X_train and Y_train in the same format as those in the tutorial and trained a model. However when I do tagger.dump(), I found these:
FILEHEADER = {
  magic: lCRF
  size: 4236
  type: FOMC
  version: 100
  num_features: 0
  num_labels: 0
  num_attrs: 0
  off_features: 0x30
  off_labels: 0x3C
  off_attrs: 0x854
  off_labelrefs: 0x106C
  off_attrrefs: 0x1080
}

LABELS = {
}

ATTRIBUTES = {
}

TRANSITIONS = {
}

STATE_FEATURES = {
}

So it seems the training data is not fed appropriately. I'm trying to figure out why. The only clue I have right now is that in the tutorial the string are unicode, while mine is utf8, do you have any suggestion based on my model dump result? Thanks you.

**Comment by Ridinstar** [comment]
@kmike Thanks for replying, never mind, it's a silly mistake in formatting training data.

**Comment by kmike** [comment]
@Ridinstar mistake in formatting training data shouldn't cause a segfault, it is a python-crfsuite bug if it does :)

**Comment by Ridinstar** [comment]
@kmike I am able to reproduce that segfault if I append no data to trainer, and I use this trainer to predict the labels of a valid test sequence.

**Comment by cortadocodes** [comment]
Training a model with no data and trying to predict with it also seems to cause a segfault:
```
import pycrfsuite


MODEL_FILENAME = 'b.crfsuite'


trainer = pycrfsuite.Trainer(verbose=False)
trainer.train(MODEL_FILENAME)

tagger = pycrfsuite.Tagger()
tagger.open(MODEL_FILENAME)

# Causes segmentation fault
tagger.tag(['hello'])
```

The following error is produced:
```
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
```
